vcstool/commands/compare.py

```diff
@@ -1,80 +1,595 @@
 import argparse
-from cgitb import reset
-from distutils.util import execute
-from gettext import find
-from prettytable import PrettyTable
 import sys
-from unittest import result
-from vcstool import clients
-from vcstool.commands.branch import BranchCommand
-from vcstool.crawler import find_repositories
+import os
+import abc
+from typing import Any, Dict, List, Optional
+from enum import IntEnum, IntFlag, auto
+
+import prettytable as pt
 
-from vcstool.executor import ansi, execute_jobs, generate_jobs, output_repositories
+from vcstool.crawler import find_repositories
+from vcstool.executor import ansi, execute_jobs, generate_jobs
 from vcstool.commands.import_ import get_repositories
 from vcstool.streams import set_streams
+from vcstool.outputs import CompareOutput
 
 from .command import add_common_arguments
 from .command import Command
 
 
 class CompareCommand(Command):
 
-    command = 'compare'
-    help = 'Compare working copy to the repository list file'
+    command = "compare"
+    help = "Compare working copy to the repository list file"
 
-    def __init__(self, args):
-        super(CompareCommand, self).__init__(args)
+    def __init__(self, args: argparse.Namespace) -> None:
+        super().__init__(args)
+        self.progress = args.progress
+        self.workers = args.workers
+        self.debug = args.debug
 
+    @classmethod
+    def get_parser(cls) -> argparse.ArgumentParser:
+        parser = argparse.ArgumentParser(description=f"{cls.help}", prog="vcs compare")
+        group = parser.add_argument_group('"compare" command parameters')
+        group.add_argument(
+            "-i",
+            "--input",
+            type=str,
+            default="workspace.repos",
+            help="Path to the repository list file for the workspace.",
+        )
+        group.add_argument(
+            "-p",
+            "--progress",
+            action="store_true",
+            default=False,
```

Comment:
Could go either way on default value here, I personally like to know that things are running, especially because when we are remote, git commands to our server will hang if we are *not* on VPN.

Reply:
Yep I get that- I just found it was a bit ugly. I also didn't find it very useful, since when the command is running normally it zips right through all the workers.

If the command hangs for more than a few seconds it's pretty clear that something is wrong, since it typically completes quite fast.